version: "2.1"
services:
  wireguard:
    image: lscr.io/linuxserver/wireguard:latest
    container_name: personal-vpn-container
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Etc/UTC
      - SERVERURL=${PERSONAL_URL}
      - SERVERPORT=${PERSONAL_VPN_PORT}
      - PEERS=${PERSONAL_PEERS}
      - PEERDNS=${PERSONAL_DNS_IPV4}
      - INTERNAL_SUBNET=${PERSONAL_SUBNET}
      - ALLOWEDIPS=0.0.0.0/0
      - PERSISTENTKEEPALIVE_PEERS=all
      - LOG_CONFS=true
    volumes:
      - /deploy/personal/wireguard:/config
      - /deploy/personal/modules:/lib/modules
    networks:
      personal-network:
        ipv4_address: ${PERSONAL_VPN_IPV4}
    ports:
      - ${PERSONAL_VPN_PORT}:${PERSONAL_VPN_PORT}/udp
    sysctls:
      - net.ipv4.conf.all.src_valid_mark=1
      - net.ipv4.ip_forward=1  # Enable IP forwarding
      - net.ipv6.conf.all.forwarding=1
    restart: unless-stopped
  unbound:
    container_name: personal-dns-container
    image: "mvance/unbound:latest"
    networks:
      personal-network:
        ipv4_address: ${PERSONAL_DNS_IPV4}
    volumes:
      - /deploy/personal/unbound/a-records.conf:/opt/unbound/etc/unbound/a-records.conf
    restart: unless-stopped
  nextcloud-aio-mastercontainer:
    networks:
      personal-network:
        ipv4_address: ${PERSONAL_NEXTCLOUD_IPV4}
    image: nextcloud/all-in-one:latest
    init: true
    restart: always
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - APACHE_PORT=11000
      - APACHE_ADDITIONAL_NETWORK=deploy_personal-network
      - APACHE_IP_BINDING=0.0.0.0
    ports:
      - "8080:8080"
  caddy:
    networks:
      - personal-network
      - nextcloud-aio
    build:
      context: ./personal/caddy
      dockerfile: Dockerfile
    restart: always
    container_name: caddy
    volumes:
      - /deploy/personal/caddy:/etc/caddy
      - /deploy/personal/caddy/certs:/certs
      - /deploy/personal/caddy/config:/config
      - /deploy/personal/caddy/data:/data
      - /deploy/personal/caddy/sites:/srv
    ports:
      - "80:80"
      - "443:443"
  homeassistant:
    container_name: homeassistant
    image: "ghcr.io/home-assistant/home-assistant:stable"
    volumes:
      - /deploy/personal/homeassistant/config:/config
      - /etc/localtime:/etc/localtime:ro
    restart: unless-stopped
    privileged: true
    networks:
      - personal-network
  redbot:
    networks:
      - personal-network
    container_name: redbot
    build:
      context: ./personal/redbot
      dockerfile: Dockerfile
      args:
        BOT_NAME: ${BOT_NAME}
  vtt:
    networks:
      - personal-network
    container_name: vtt
    hostname: vtt
    build:
      context: ./personal/vtt
      dockerfile: Dockerfile
      args:
        FOUNDRY_USERNAME: ${FOUNDRY_USERNAME}
        FOUNDRY_PASSWORD: ${FOUNDRY_PASSWORD}
    volumes:
      - /deploy/personal/vtt12/data:/root/.local/share
networks:
  personal-network:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: ${PERSONAL_IPAM_SUBNET}/24
  nextcloud-aio:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: ${NEXTCLOUD_IPAM_SUBNET}/24
volumes:
  mydata:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
